packages/e2e-tests/src/specs/editor/shopping/woocommerce.js

```diff
@@ -0,0 +1,104 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+
+// eslint-disable-next-line import/no-extraneous-dependencies -- in package.json
+import { matchers } from 'jest-json-schema';
+
+import {
+  createNewStory,
+  withExperimentalFeatures,
+  publishStory,
+  activatePlugin,
+  deactivatePlugin,
+  takeSnapshot,
+} from '@web-stories-wp/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import * as schema from './schema.json';
+
+// eslint-disable-next-line jest/require-hook -- matchers only for this set of tests
+expect.extend(matchers);
+
+describe('Shopping', () => {
+  describe('Shopping schema', () => {
+    const WPVersion = process.env?.WP_VERSION || 'latest';
+
+    withExperimentalFeatures(['shoppingIntegration']);
+
+    beforeAll(async () => {
+      if ('latest' === WPVersion) {
+        await activatePlugin('woocommerce');
+      }
+    });
+
+    afterAll(async () => {
+      if ('latest' === WPVersion) {
+        await deactivatePlugin('woocommerce');
+      }
+    });
+
+    if (WPVersion !== 'latest') {
+      //eslint-disable-next-line jest/expect-expect, jest/no-focused-tests
+      it.only('minimum WordPress requirement not met', () => {});
+    }
+
+    it('should should match a valid schema', async () => {
+      await createNewStory();
+      await expect(page).toClick('[aria-controls="library-pane-shopping"]');
+      await expect(page).toClick('[aria-label="Product"]');
+      await page.keyboard.press('ArrowDown');
+      await page.keyboard.press('Enter');
+      await expect(page).toClick('button', { text: 'Insert product' });
+      await takeSnapshot(page, 'insert product');
+      await publishStory();
+
+      // open story dev tools
+      await page.keyboard.down('Meta');
+      await page.keyboard.down('Alt');
+      await page.keyboard.down('Shift');
+      await page.keyboard.down('J');
+
+      await takeSnapshot(page, 'shopping schema');
+
+      // pull story data
+      const state = JSON.parse(
+        await page.$eval(
+          'textArea[class^="devTools__Textarea"]',
+          (el) => el.value
+        )
+      );
+      const products = [];
+      state.pages.forEach((page) => {
+        page.elements.forEach((element) => {
+          const { type, product } = element;
+          //eslint-disable-next-line jest/no-conditional-in-test -- only add products
+          if (type === 'product') {
+            products.push(product);
+          }
+        });
+      });
+
+      expect(products).toHaveLength(1);
+      expect(products[0]).toMatchSchema(schema);
```

Comment:
The product element data does not necessarily map to the product schema 1:1.

What's important is that the final product output when previewing a story is valid.

So instead of opening devtools, let's preview the story, grab the product config from the amp-story-shopping config and validate that one.

Reply:
Updated to pull from the preview